src/docs/cookie-clicker-swiftui.md

```diff
@@ -0,0 +1,314 @@
+
+Create a state variable at the top of the definition of `ContentView`. We make it `private` because it is internal to the view, and we give it a default value of `false` which implicitly makes `showGreeting` a variable that holds a `Bool` (Now is the time to ask your Coach questions).
+
+```swift
+struct ContentView: View {
+    @State private var showGreeting = false
+    
+    var body: some View {
+    <...>
+```
+
+We are going to use this to configure what the view displays.
+
+### 9. Configure the view to display according to the state
+
+Change the string that is displayed in the `Text` view depending on this state:
+
+```swift
+Text(showGreeting ? "Hello CodeBar!" : "")
+```
+
+When `showGreeting` is `true` the label will display the greeting, otherwise it will be empty.
+
+### 10. Use the Button's action to change the state
+
+When you dragged the button onto the canvas it created an empty `action` This is a closure, and when the button is tapped it will perform whatever code is written in this closure. We are going to use this to toggle the `showGreeting` state.
+
+Edit the Button code look like this:
+
+```swift
+Button(action: { self.showGreeting.toggle() })
+```
+
+The `ContentView` struct should now look like this:
+
+```swift
+struct ContentView: View {
+    @State private var showGreeting = false
+    
+    var body: some View {
+        VStack {
+            Text(showGreeting ? "Hello CodeBar!" : "")
+            Button(action: { self.showGreeting.toggle() }) {
+                Text("Button")
+            }
+        }
+    }
+}
+```
+
+### 11. Run the app in the Simulator or by using the Canvas's live preview feature
+
+As you tap the button, the greeting appears and disappears!
+
+/giphy Celebrate
+
+
+This is not trivial. You have learned an important part of developing using SwiftUI. You create state, you write your view to respond to the state, and you have actions that change the state. The framework takes care of redrawing the view when the state changes.
+
+If you have done any iOS development before, or even the other CodeBar tutorial, you can see how little code is needed. There are no `ViewController`s. There are no `Storyboards`. You write code and see the results immediately.
+
+ 
+## Make the Counter
+
+There are two major tasks to get the make the counter app. The functionality - getting the counter to increment on button clicks and showing the new number. The UI, getting an app that looks nice and is easy to use. This tutorial will create the functionality first and then move on to the UI. You may choose to tackle the tasks in the alternate order. Have a brief discussion with your Coach about the pros and cons of each approach.
+
+### 12. Create the Counter functionality
+
+We are a good way towards this, we already have a `Button` with an action, and a `Text` view that displays something. 
+
+Change the state variable to hold a number instead of a `Bool`
+
+```swift
+@State private var count = 0
```

Comment:
might be a stupid question, but what does @state add to a variable?

Reply:
Ah. That's a really good point. @State is important. It's a property modifier that the `body` of the view uses to know that when it changes the view should redraw. If you didn't understand that (I know you'll take that phrase in the way I mean it), then I haven't explained it properly. I'll see if I can do an edit to make it clearer.

Reply:
Thank you 🎉